core/controllers/acl_decorators_test.py

```diff
@@ -3529,3 +3532,185 @@ def test_that_no_excpetion_is_raised_when_valid_vm_access(self):
+        self.login(self.author_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json('/mock/%s' % self.suggestion_id_01)
+        self.assertEqual(response['suggestion_id'], self.suggestion_id_01)
+        self.logout()
+
+    def test_admin_can_update_suggestion(self):
+        self.login(self.admin_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json('/mock/%s' % self.suggestion_id_01)
+        self.assertEqual(response['suggestion_id'], self.suggestion_id_01)
+        self.logout()
+
+    def test_reviewer_can_update_suggestion(self):
+        self.login(self.reviewer_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json('/mock/%s' % self.suggestion_id_01)
+        self.assertEqual(response['suggestion_id'], self.suggestion_id_01)
+        self.logout()
+
+    def test_non_author_cannot_update_suggestion(self):
+        self.login(self.user_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json(
+                '/mock/%s' % self.suggestion_id_01, expected_status_int=401)
+        self.assertEqual(
+            response['error'],
+            'You are not allowed to update the suggestion.')
+        self.logout()
+
+    def test_guest_cannot_update_suggestion(self):
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json(
+                '/mock/%s' % self.suggestion_id_01, expected_status_int=401)
+        self.assertEqual(
+            response['error'],
+            'You must be logged in to access this resource.')
+
+    def test_suggestions_with_invalid_suggestion_id_cannot_be_updated(self):
+        self.login(self.reviewer_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json(
+                '/mock/%s' % 'suggestion-id',
+                expected_status_int=400)
+        self.assertEqual(
+            response['error'], 'Invalid format for suggestion_id. ' +
+            'It must contain 3 parts separated by \'.\'')
+        self.logout()
+
+    def test_non_existent_suggestion_cannot_be_updated(self):
+        self.login(self.reviewer_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            self.get_json(
+                '/mock/%s' % 'exploration.exp1.' +
+                'WzE2MTc4NzExNzExNDEuOTE0XQ==WzQ5NTs',
+                expected_status_int=404)
+        self.logout()
+
+    def test_non_translation_suggestions_are_rejected(self):
+        self.login(self.admin_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json(
+                '/mock/%s' % self.suggestion_id_02, expected_status_int=401)
+        self.assertEqual(
+            response['error'],
+            'You are not allowed to update the suggestion.')
+        self.logout()
```

Comment:
Added a test to verify that question suggestions are not included with this decorator.
Reference: https://github.com/oppia/oppia/pull/12424/files#r612894871
/cc @BenHenning

Reply:
Thanks for adding this!